### PR TITLE
Convert `ClipboardButton` to TypeScript

### PR DESCRIPTION
**Pull request by sunyatasattva**
## What?
This PR converts the deprecated `ClipboardButton` component to TypeScript.

## Why?
Even if it is deprecated, it was still on the list in #35744. It was easy enough, so I think it's better to have even deprecated components converted to TypeScript.

### CODE REVIEW

**Review by mirka — COMMENTED** [review]
Looks good, thank you! 🎉

We can merge once you add a quick [changelog](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/CHANGELOG.md) 👍

**Comment by sunyatasattva** [comment]
@mirka Changelog pushed 😊  Thanks for the review!

File: packages/components/CHANGELOG.md [hunk 1]
```diff
@@ -22,9 +22,13 @@
...
 
 -   `DropdownMenu` v2: Tweak styles ([#50967](https://github.com/WordPress/gutenberg/pull/50967), [#51097](https://github.com/WordPress/gutenberg/pull/51097)).
--   `DropdownMenu` v2: change default placement to match the legacy `DropdownMenu` component  ([#51133](https://github.com/WordPress/gutenberg/pull/51133)).
+-   `DropdownMenu` v2: change default placement to match the legacy `DropdownMenu` component ([#51133](https://github.com/WordPress/gutenberg/pull/51133)).
 -   `DropdownMenu` v2: Render in the default `Popover.Slot` ([#51046](https://github.com/WordPress/gutenberg/pull/51046)).
 
+### Internal
+
+-   `ClipboardButton`: Convert to TypeScript ([#51334](https://github.com/WordPress/gutenberg/pull/51334)).
+
```
Comment (mirka) [inline packages/components/CHANGELOG.md]: We need to move this to the "Unreleased" section at the top.

Comment (sunyatasattva) [reply packages/components/CHANGELOG.md]: Done :)

**Review by mirka — APPROVED** [review]
Perfect, thanks!

**Pull request merged**